Three/include/CGAL/Three/Scene_item_with_properties.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2016  GeometryFactory Sarl (France)
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+// Author(s)     : Maxime GIMENO
+
+#ifndef SCENE_ITEM_WITH_PROPERTIES_H
+#define SCENE_ITEM_WITH_PROPERTIES_H
+namespace CGAL
+{
+namespace Three {
+  class Scene_item;
+
+//! Base class to allow an item to copy properties from another.
+class Scene_item_with_properties {
+public:
+  virtual ~Scene_item_with_properties(){}
+ //!Copies properties from another Scene_item
+ virtual void copyProperties(Scene_item*){}
```

Comment:
You can use:
```C++
  virtual void copyProperties(Scene_item*) = 0; // pure virtual method
```
That will avoid that somebody derives from `Scene_item_with_properties` and forgets to override `copyProperties`, or overrides it with the wrong signature.